File: app/src/main/java/ru/mamykin/foreignbooksreader/ui/adapters/MainViewPagerAdapter.java
package ru.mamykin.foreignbooksreader.ui.adapters;

import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Адаптер для ViewPager главного экрана
 */
public class MainViewPagerAdapter extends FragmentStatePagerAdapter {
    private List<Fragment> fragmentsList = new ArrayList<>();
    private List<String> titlesList = new ArrayList<>();

    public MainViewPagerAdapter(FragmentManager manager) {
        super(manager);
    }

    @Override
    @Nullable
    public Fragment getItem(int position) {
        return fragmentsList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentsList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titlesList.get(position);
    }

    public void addFragment(Fragment fragment, String title) {
        fragmentsList.add(fragment);
        titlesList.add(title);
    }

    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }
}

File: app/src/main/java/ru/mamykin/foboreader/ui/main/MainViewPagerAdapter.kt
package ru.mamykin.foboreader.ui.main

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.view.PagerAdapter

/**
 * Адаптер для ViewPager главного экрана
 */
class MainViewPagerAdapter(manager: FragmentManager) : FragmentStatePagerAdapter(manager) {

    private val fragments = mutableListOf<Fragment>()
    private val titles = mutableListOf<String>()

    override fun getItem(position: Int): Fragment = fragments[position]

    override fun getCount(): Int = fragments.size

    override fun getPageTitle(position: Int): CharSequence = titles[position]

    override fun getItemPosition(obj: Any): Int = PagerAdapter.POSITION_NONE

    fun addFragment(fragment: Fragment, title: String) {
        fragments.add(fragment)
        this.titles.add(title)
    }
}